tests/unittests/GradCheckTest.cpp

```diff
@@ -236,25 +236,26 @@ TEST_P(InterpreterGrad, gradientCheckMatMul) {
 
 TEST_P(InterpreterGrad, gradientCheckBatchedReduceAdd) {
   PlaceholderBindings Bindings;
-  size_t BatchSize = 40;
-  size_t NumCols = 12;
+  size_t BatchSize = 4;
+  size_t NumRows = 3;
+  size_t NumCols = 5;
 
   auto &Mod = EE_.getModule();
   Function *F = Mod.createFunction("main");
 
-  auto *A = Mod.createPlaceholder(ElemKind::FloatTy, {BatchSize, NumCols}, "A",
-                                  false);
+  auto *A = Mod.createPlaceholder(ElemKind::FloatTy,
+                                  {BatchSize, NumRows, NumCols}, "A", false);
 
-  auto *Exp =
-      Mod.createPlaceholder(ElemKind::FloatTy, {1, NumCols}, "exp", false);
+  auto *Exp = Mod.createPlaceholder(ElemKind::FloatTy, {NumRows, NumCols},
+                                    "exp", false);
 
-  TypeRef Ty = Mod.uniqueType(ElemKind::FloatTy, {1, NumCols});
+  TypeRef Ty = Mod.uniqueType(ElemKind::FloatTy, {NumRows, NumCols});
   Node *BRA = F->createBatchedReduceAdd("BRA", Ty, A, 0 /*axis*/);
```

Comment:
Consider adding a test case with `axis` not equal to `0` to test the case I commented about above.